docs/konnectors_workflow_example.md

```diff
@@ -0,0 +1,373 @@
+```
+```json
+{
+  "data": [
+    {
+      "type": "io.cozy.konnectors",
+      "id": "io.cozy.konnectors/trainlines"
+    }
+  ]
+}
+```
+
+10 - SettingsApp runs the konnector the first time
+
+```http
+POST /jobs/queue/konnector
+```
+```json
+{
+  "data": {
+    "attributes": {
+       "options": {
+       "priority": 10,
+       "timeout": 180,
+       "max_exec_count": 1
+      }
+    },
+    "arguments": {
+      "konnector": "io.cozy.konnectors/trainlines",
+      "account": "123-account-id-123",
+      "folderToSave": "123-selected-folder-id-123"
+    }
+  }
+}
+```
+```http
+HTTP/1.1 200 OK
+```
+```json
+{
+  "data": {
+    "id":"789-job-id-789",
+    "type":"io.cozy.jobs",
+    "attributes": {
+      "worker": "konnector",
+      "options": {
+        "priority": 3,
+        "timeout": 60,
+        "max_exec_count": 3
+      },
+      "arguments": {
+        "konnector": "io.cozy.konnectors/trainlines",
+        "account": "123-account-id-123",
+        "folderToSave": "123-selected-folder-id-123"
+      },
+      "state": "running",
+      "try_count": 1,
+      "queued_at": "2016-09-19T12:35:08Z",
+      "started_at": "2016-09-19T12:35:08Z",
+      "errors": [],
+      "output": {}
+    }
+  }
+}
+```
+
+11 - SettingsApp follow konnector status to check if all is properly configured.
+It uses the [realtime](./realtime.md) to subscribes to changes on `789-job-id-789` and is therefore informed of it's status / errors / ect ...
+
+
+**TODO** Look at current konnectors sources to defines a protocol between konnectors and SettingsApp to display the nice progress modal.
+  - [ ] 250 events imported
+  - [ ] 150 / 3500 contacts importing
+  - [ ] ...
+
+**TODO** there should be some persistence for jobs error / status
+
+
+12 - SettingsApp creates a trigger to setup the konnector recurence
+
+```http
+POST /jobs/io.cozy.triggers
+```
+```json
+{
+  "data": {
+    "attributes": {
+      "type": "@cron",
+      "arguments": "0 0 0 0 1 1 ",
+      "worker": "konnector",
+      "worker_arguments": {
+        "konnector": "trainlines",
+        "account": "5165621628784562148955",
+        "folderToSave": "877854878455",
+      }
+    }
+  }
+}
+```
+
+If the user wants to use several account, Settings can setup several triggers for the same konnector & various accounts.
+
+
+### Relations and DELETE CASCADE
+
+- When a konnector is deleted, its triggers should be deleted
+- When an account is deleted, its triggers should be deleted
+
+
+**TODO** Should stack validate konnector value on trigger creation / to be extended to all workers pre-validating worker arguments ?
+
+
+## Konnector Worker specs
+
+- Start the konnector through Rkt, passing as ENV variables :
+    - COZY_CREDENTIALS  security token to communicate with Cozy
+    - COZY_FIELDS       JSON-encoded worker_arguments
+    - COZY_DOMAIN       the starting instance domain
+
+- Read konnector status from process stdout and emit them as
+  `io.cozy.jobs.event` on the realtime hub.
```

Comment:
I we want that we should be a little more precise about what constitute an "event" on the stdout (simple strings separated by newlines ? json object separated by newlines ? etc..)